test/integration/targets/tower_group/tasks/main.yml

```diff
@@ -22,6 +22,19 @@
     that:
       - "result is changed"
 
+- name: Check changed status is accurate
+  tower_group:
+    name: Some Group
+    inventory: Demo Inventory
+    state: present
+    source: ec2
```

Comment:
This fails if `source: ` is specified. taking out the source line; even just from the 2nd tower_group call give the expected/desired result. 

Should I remove the `source: ec2` lines from both as `source` is legacy anyway?

Reply:
What about checking the tower_cli.__version__ in the tower_group module. And removing the parameters if its on the newer API version?

```
if int(v[0]) >= 3 and int(v[1]) >= 2:
    module.params.pop('source')
    module.params.pop('source_regions')
    ...
    ...
```

Reply:
I think it's fairly normal for other modules to use `LooseVersion`, so that's a possibility. However, if you do this, I don't think it's reasonable to test it in CI, because that's only using the current version of tower-cli, which is specific to the v2 API.